storage/s3.go

```diff
@@ -87,13 +80,16 @@ func parseEndpoint(endpoint string) (urlpkg.URL, error) {
 }
 
 // NewS3Storage creates new S3 session.
-func newS3Storage(opts Options, sessProvider func() *session.Session) (*S3, error) {
+func newS3Storage(ctx context.Context, opts Options, sessionProvider *s3Session) (*S3, error) {
```

Comment:
```suggestion
func newS3Storage(ctx context.Context, opts Options) (*S3, error) {
```

Since sessionProvider is global, it can be directly accessed.